<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>UGA Talk</title>
		<meta name="description" content="UGA Talk">
		<meta name="author" content="Scott Chamberlain">
		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="css/reveal.min.css">
		<link rel="stylesheet" href="css/theme/solarized.css" id="theme">
		<link rel="stylesheet" href="css/style.css">

		<link href='http://fonts.googleapis.com/css?family=Source+Sans+Pro' rel='stylesheet' type='text/css'>

		<!-- <link rel="stylesheet" href="http://netdna.bootstrapcdn.com/bootstrap/3.0.3/css/bootstrap.min.css"> -->
		<link rel="stylesheet" href="http://netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css">
		<link rel="stylesheet" href="lib/css/font-awesome-4.2.0/css/font-awesome.css">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">
		<!-- <link rel="stylesheet" href="lib/css/foundation.css"> -->
		<!-- <link rel="stylesheet" href="lib/css/font-awesome.min.css"> -->

		<!-- If the query includes 'print-pdf', use the PDF print sheet -->
		<script>
			document.write( '<link rel="stylesheet" href="css/print/' + ( window.location.search.match( /print-pdf/gi ) ? 'pdf' : 'paper' ) + '.css" type="text/css" media="print">' );
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->

		<style type="text/css">
			#rotate_text {
				-webkit-transform: rotate(-90deg);
				-moz-transform: rotate(-90deg);
				transform: rotate(-90deg);
				position: absolute;
				vertical-align: middle;
				filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
			}

			ul.a {list-style-type: circle; margin: 0; padding: 0;}
			ul.b {list-style-type: square; margin-left: 0; padding-left: 0;}
		</style>
	</head>

	<body>

		<div class="reveal">

	<!-- Any section element inside of this container is displayed as a slide -->
	<div class="slides">
		<section id="intro">
				<img src="img/ropensci_main.png" alt="ropensci" />
				<h2><font color="#B8172F">Cultivating open and reproducible science</font></h2><br>
				<p class="grey">Scott Chamberlain (@recology_)
					<p class="grey">UC Berkeley
				<p class="grey">recology.info/talks/uga<br><br>
				<center>Supported by:</center>
				<img src="img/sloan_logo2.png" alt="sloan" />
		</section>

		<section id="notes">
			<center><img src="img/notesfromnature.png" width="1100"></center>
		</section>

		<section id="sensors">
			<center><img src="img/sensors.png" width="1100"></center>
		</section>

		<section id="manual_data">
			<center><img src="img/ropensci_data.png" width="1100"></center>
		</section>

		<section id="messy_data">
			<center><img src="img/esa_archives.png" width="1200"></center>
		</section>

		<!-- <section id="embargo">
			<center><img src="img/dryad_embargo_red.png" width="1100"></center>
		</section> -->

		<section id="data_hard">
			<h1 align="left">These data are hard to get</h1>
			<h2><i class="fa fa-lock fa-5x" style="color:#b22d28"></i></h2>
		</section>

		<section id="we_need">
			<center><h1 align="left">We need...</h2></center>
				<br><br>
				<center><div class="row">
					<font color="green"><h2><i class="fa fa-unlock fa-5x"></i></font></h2></font>
				</div></center>
		</section>

		<section id="cultural_barriers">
			<h1>Cultural barriers</h1>
			<br><br>
			<ul style="float:left;">
				<h2>Lack of incentives (carrots)</h2>
				<h2>Lack of pressure (sticks)</h2>
				<h2>Getting scooped (<i class="fa fa-spoon"></i>)</h2>
				<h2>Takes too much time! (<i class="fa fa-clock-o"></i>)</h2>
			</ul>
		</section>

		<section id="nfs">
		 	<h2><font color="#C3C4C7">Instructions for preparation of the Biographical Sketch have been revised to rename the <font color="black">"Publications" section to "Products"</font> and amend terminology and instructions accordingly. This change makes clear that products may include, but are not limited to, publications, <font color="black">data sets, software, patents, and copyrights.</font></font></h2>
			<br><br><br>
			<p><a href="http://www.nsf.gov/pubs/2013/nsf13004/nsf13004.jsp?WT.mc_id=USNSF_109">Issuance of a new NSF Proposal & Award Policies and Procedures Guide (October 4th)</a></p>
		</section>

		<section id="plos">
			<h2><font color="#C3C4C7">the Data Policy states the <font color="black">‘minimal dataset’</font> consists “of the <font color="black">dataset used to reach the conclusions</font> drawn in the manuscript with related metadata and methods, and any additional data required to replicate the reported study findings in their entirety. This does not mean that authors must submit all data collected as part of the research, but that they must provide the <font color="black">data that are relevant to the specific analysis presented in the paper.</font></font></h2>
			<br>
			<p><a href="http://www.plosone.org/static/policies#sharing">PLOS Editorial and Publishing Policies</a></p>
		</section>

		<section id="why_open">
				<h1> <font color="green"><i class="fa fa-unlock"></i></font>&nbsp;Why Open?</h1>
		</section>

		<section id="why_open_public">
			<center>
				<font color="green"> <i class="fa fa-unlock fa-3x">&nbsp;Why Open?</i></font>
				<br><br><br><br>
				<h2>Most research publicly funded</h2>
				<br><br>
				<center><i class="fa fa-dollar fa-5x"></i>
		</section>

		<section id="why_open_pace">
			<font color="green"> <i class="fa fa-unlock fa-3x">&nbsp;Why Open?</i></font>
			<br><br><br><br>
			<h2>To increase the pace of science</h2>
			<br><br>
			<i class="fa fa-rocket fa-5x"></i>
		</section>

		<section id="why_open_citations">
			<center>
				<font color="green"> <i class="fa fa-unlock fa-3x">&nbsp;Why Open?</i></font>
				<br><br>
				<h2> Sharing data increases citations</h2>
				<center><img src="img/piwowar.png" width="700"></center>
			<a href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0000308" style="float:right;">Piwowar et al. 2007</a>
		</section>

		<section id="reproducible">
				<h1> Openness <font color="green"><i class="fa fa-unlock"></i></font> facilitates reproducibility <font color="blue"><i class="fa fa-refresh"> </i></font></h1>
		</section>

		<section id="why_reproducible_selfish">
			<font color="blue"><i class="fa fa-refresh fa-3x">&nbsp;Why Reproducible?</i> </font>
			<br><br><br><br>
			<h2>For yourself!</h2>
			<br><br>
			<i class="fa fa-user fa-5x"></i>
			<!-- <h2 style="float:left;"> <i class="fa fa-code"></i> for others (if reproducible, more confidence in results) </h2> -->
		</section>

		<section id="why_reproducible_mistakes">
			<font color="blue"><i class="fa fa-refresh fa-3x">&nbsp;Why Reproducible?</i> </font>
			<br><br><br>
			<h2>To avoid mistakes</h2>
			<br><br>
			<table cellpadding="40" cellspacing="40">
			<tr>
			<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td>
				<td>		<center><p>Reinhart-Rogoff<img src="img/excel.png" width="500"></center>
				<a href="http://www.nytimes.com/2013/04/19/opinion/krugman-the-excel-depression.html?_r=0" style="float:left;">excel</a></td>
			<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td>
				<td>		<center><p>Chronic fatigue syndrome-XMRV<img src="img/judy.jpg" width="300"></center>
				<a href="http://retractionwatch.com/2011/12/22/chronic-fatigue-syndrome-xmrv-paper-retracted-by-science-completely-this-time/" style="float:left;">cfs</a></td>
			</tr>
			</table>
		</section>

		<!-- <section id="easier">
			<font size="52"><h1 align="left">Or maybe you just want science to be easier</h1></font>
		</section> -->

		<section id="whats_needed">
			<font size="52"><h1 align="left">What's needed?</h1></font>
		</section>

		<section id="toolset">
			<br>
			<img src="img/tools.png" alt="toolset" width="1200" />
		</section>

		<section id="reproducible_workflow">
			<font size="52"><h1 align="left"><i class="fa fa-arrow-circle-right"></i>&nbsp;A reproducible workflow</h1></font>
		</section>

		<section id="not_these">
			<center><h2>Not these!</h2></center>
			<img src="img/closedtools.png" width="1000">
		</section>

		<section id="these">
			<center><h2>These!</h2></center>
			<img src="img/opentools.png" width="1000">
		</section>

		<section id="learning_curve">
			<center><h2>There's a learning curve, but...</h2></center>
			<img src="img/tasks.jpg" width="1000">
			<a href="http://www.globalnerdy.com/2012/04/24/geeks-and-repetitive-tasks/">link</a>
		</section>

		<section id="workflows">
		<center><h2> Workflows side by side</h2></center>
			<table cellpadding="40" cellspacing="20">
				<tr>
	<td>&nbsp;&nbsp;</td>
		<td width="50%"> <p>Our workflow now
				<ul>
					<font size="6">
					<li>Browser</li>
					<li>Excel</li>
					<li>SAS</li>
					<li>SigmaPlot</li>
					<li>Word</li>
					<li>Endnote</li>
				</font>
				</ul> </td>
		<td>&nbsp;&nbsp;</td>

		 <td width="50%">	<p>What it could be
			<ul>
				<li>R (or Python)</li>
				<li>+ a few open source things (<a href="http://daringfireball.net/projects/markdown/">Markdown</a>, or <a href="http://www.latex-project.org/">LaTeX</a> )</li></ul></td>
	<td>&nbsp;&nbsp;</td>
		</tr>
		<tr>
<td>&nbsp;&nbsp;</td>
				<td><p> <i class="fa fa-money"></i> Cost = $$$$$$$<br>
					<p> <i class="fa fa-unlock"></i> Open? = Nope
					<p> <i class="fa fa-refresh"></i> Reproducible? = Nope</td>
<td>&nbsp;&nbsp;</td>
				<td><p> <i class="fa fa-money"></i> Cost = 0<br>
					<p> <i class="fa fa-unlock"></i> Open? = Yes!
					<p> <i class="fa fa-refresh"></i> Reproducible? = Yep</td>

		</tr>
	</table>
		</section>

		<section id="perfect">
			<center><h2>A perfect marriage</h2></center>
			<br><br>
			<center>
				<font color="red" size="12"><i class="fa fa-refresh fa-5x">&nbsp;</i></font>
				<font color="green" size="12"> <i class="fa fa-unlock fa-5x">&nbsp;</i></font>
				<font color="blue" size="12" class="fragment"><i class="fa fa-cloud fa-5x">&nbsp;</i></font>
			</center>
		</section>

		<section id="dataweb">
			<h2>Data is increasingly on the web <font color="blue"><i class="fa fa-cloud"></i></font></h2>
			<img src="img/data.png" width="1000">
		</section>

		<section id="apis1">
			<br><br>
			<h2>API: Application Programming Interface</h2>
			<br><br><br><br>
			<h2><i class="fa fa-link fa-3x"></i>&nbsp;&nbsp;</h2>
		</section>

		<section id="apis2">
			<br><br>
			<h2> <font color="red">Reproducibly</font> plug data from the web into your science</h2>
		</section>

		<section id="ropensci">
			<br><br><br>
			<center><img src="img/ropensci_main.png"></center>
			<br><br>
			<center><img src="img/sloan_logo2.png"></center>
		</section>

		<section id="ropensci_team">
			<h2>How rOpenSci got started</h2>
			    <div>
			    <img src="img/core_team.png">
			    </div>
					<br><br><br>
			    <h3>formed from ad-hoc conversation over Twitter. Now a worldwide community of researchers <br><br> <a href="http://ropensci.org/community/#community" title="rOpenSci Community" target="_blank">http://ropensci.org/community</a>.</h3>
		</section>

		<!-- <section id="ropensci_does">

		</section> -->

		<section id="workflow">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_acquisition">
			<br><br><br>
			<center>
				<h2><font color="blue"><strong>Data acquisition</strong> <i class="fa fa-cloud-download"></i></font> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_manip">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> <font color="blue"><strong>data manipulation/analysis/viz</strong> <i class="fa fa-bar-chart"></i></font> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_writing">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> <font color="blue"><strong>writing</strong> <i class="fa fa-pencil"></i></font> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_publish">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> <font color="blue"><strong>publish</strong> <i class="fa fa-cloud-upload"></i></font></h2>
			</center>
		</section>

		<section id="packages">
			<h2>rOpenSci packages</h2>

			<h3><a href="http://ropensci.org/packages/">ropensci.org/packages</a> </h3>

			<table cellpadding="20" cellspacing="20">
				<tr>
	<!-- <td>&nbsp;</td> -->
		<td>	<h3> <font color="red"> <i class="fa fa-table">&nbsp;Data</i> </font> </h3>
			<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/taxize">taxize</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rgbif">rgbif</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rbison">rbison</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/treebase">treebase</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rfishbase">rfishbase</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rWBclimate">rWBclimate</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/EML">EML</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/RNeXML">RNeXML</a></h3> </li>
				</ul>
		</td>
		<td>&nbsp;</td>
		<td>
				<h3> <font color="blue"><i class="fa fa-book">&nbsp;Literature</font></i> </h3>
				<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/rplos">rplos</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/RMendeley">RMendeley</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rmetadata">rmetadata</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rentrez">rentrez</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/fulltext">fulltext</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/bmc">bmc</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/elife">eLife</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rcrossref">rcrossref</a></h3> </li>
				</ul>
</td>
<td>&nbsp;</td>
			<td>
			 <h3> <font color="black"><i class="fa fa-bar-chart-o">&nbsp;Altmetrics</font></i> </h3>
					<br>
					<ul>
						<li> <h3><a href="https://github.com/ropensci/alm">alm</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/rAltmetric">rAltmetric</a></h3> </li>
					</ul>
			</td>
		<td>&nbsp;</td>
			<td>
					<h3><font color="green"><i class="fa fa-bars">&nbsp;Publishing</font></i></h3>
					<br>
					<ul>
						<li> <h3><a href="https://github.com/ropensci/rfigshare">rfighsare</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/EML">EML</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/zenodo">zenodo</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/dvn">dvn</a></h3> </li>
					</ul>
			</td>
			</table>
		</section>

		<section id="demos">
			<h1>Code demos</h1>
			<br><br>
			<ul>
				<h2>The diversity of projects</h2>
				<h2>How easy it is</h2>
				<br>
				<h2>Image examples at scale</h2>
			</ul>
		</section>

		<!-- <section>
			<h1>Hold up - Why would I want to get data programatically?</h1>
		</section>

		<section>
			<h2 style="float:left;"> <i class="fa fa-code"></i> Science is just easier </h2>
			<br><br><br>
			<h2 style="float:left;"> <i class="fa fa-code"></i> You can reproduce your work </h2>
			<br><br><br>
			<h2 style="float:left;"> <i class="fa fa-code"></i> Others can reproduce your work </h2>
		</section> -->

		<section id="start_data">
			<br><br><br>
			<h1><font color="red"><i class="fa fa-table">&nbsp;Data</font></i></h1>
		</section>

		<section data-markdown id="rgbif">
			<script type="text/template">
			## Mapping biodiversity data - <font color="red">rgbif</font>

			```r
			library(rgbif)
			key <- name_backbone(name='Danaus plexippus', kingdom='animals')$speciesKey
			out <- occ_search(taxonKey=key, limit=300, return='data')
			gbifmap(out)
			```

			<img src="code/figure/rgbif.png" alt="rwbclimate" width="1100" />
			</script>
		</section>

		<section data-markdown id="rwbclimate">
			<script type="text/template">
			## Projected climate data - <font color="red">rWBclimate</font>

			```r
			library('rWBclimate'); library('ggplot2')
			country_dat <- get_historical_temp(c("USA", "MEX", "CAN", "BLZ"), "year")
			ggplot(country_dat, aes(x = year, y = data, group = locator)) +
				theme_bw(base_size=18) +
				geom_point() +
				geom_path() +
				labs(y="Average annual temperature", x="Year") +
				stat_smooth(se = F, colour = "black") +
				facet_wrap(~locator, scale = "free")
			```

			<img src="code/figure/rwbclimate2.png" alt="rwbclimate" width="1000" />
			</script>
		</section>

		<section id="rentrez_david">
			<h1> rentrez - an R client for ENTREZ </h1>
			<br>
			<img src="img/community/david_winter.png" alt="david winter" />
		</section>

		<section data-markdown id="rentrez_data">
			<script type="text/template">
			## Get FASTA format sequence data - <font color="red">rentrez</font>

			```r
			library(rentrez)
			res <- entrez_search(db="nuccore", "Gastropoda[Organism] AND COI[Gene]")
			seq <- entrez_fetch(db="nuccore", id=res$ids[1], rettype="fasta")
			cat(seq)
			```
<br><br>

			```
			>gi|685146762|emb|HG932315.1| Lymnaea stagnalis mitochondrial partial COI gene for cytochrome oxidase subunit 1, specimen voucher SNSD Moll S955
			TATATAATTTTTGGTGTTTGATGTGGGTTAGTTGGGACAGGTTTATCCTTACTAATTCGTTTAGAGTTAG
			GTACGTCTATAGTTTTAATTGATGAGCATTTTTACAATGTTATTGTTACAGCACATGCTTTTGTTATGAT
			TTTTTTTATAGTTATACCTATAATGATTGGGGGTTTTGGAAATTGAATAGTACCTTTATTAATTGGGGCT
			CCAGATATAAGTTTCCCTCGAATAAATAATATAAGTTTTTGGTTACTCCCTCCTTCCTTTGTTCTTCTTT
			TATGTTCCAGTATAGTTGAAGGTGGGGTAGGTACAGGATGAACAGTATACCCTCCTCTAAGAGGTCCTAT
			TGCTCATAGTGGTTCATCTGTAGATTTAGCTATTTTTTCATTACATTTAGCAGGTTTATCCAGAATTTTA
			GGTGCTATTAATTTTATTACAACTATTTTTAACATACGATCACCGGGTATAACTTTAGAACGAATAAGAT
			TATTTGTTTGGTCTGTGTTAGTTACAGCTTTTTTACTTTTACTTTCTTTACCTGTTTTNGCTGGGGCTAT
			TACAATACTTTTAACTGACCGAAATTTTAACACTACTTTTTTTGACCCAGCTGGAGGTGGGGATCCTATT
			TTATATCAACATTTGTTT
			```
			</script>
		</section>

		<section id="neotoma_simon">
			<h1> neotoma - an R client for the Neotoma Paleoecology database </h1>
			<br>
			<img src="img/community/simon_goring.png" alt="simon goring" />
		</section>

		<section data-markdown id="neotoma">
			<script type="text/template">
			## Get XXX - <font color="red">neotoma</font>

			```r
			library('neotoma')

			```
<br><br>

			```

			```
			</script>
		</section>

		<section id="taxize_people">
			<h1>taxize <br> <small>all things taxonomy</small> </h1>
			<br><br>
			<img src="img/community/ben_mawrick.jpg" alt="ben marwick" width="300" />
			<img src="img/community/ed_szocs.png" alt="ed" width="300"  />
			<img src="img/community/scott.png" alt="Scott Chamberlain" width="300"  />
			<img src="img/community/karthik.png" alt="karthik ram"  width="300" />
			<br><h3>
				and many more...
			</h3>
		</section>

		<section data-markdown id="taxize_approach">
			<script type="text/template">
				# taxize summary


				- ## Data sources, Lots! currently 20
				- ## Verbs (ish)
				  - ### `classification()`
				  - ### `children()`
				  - ### `downstream()`
				  - ### `comm2sci()`, `sci2comm()`
				  - ### `resolve()`
				  - ### `synonyms()`
			</script>
		</section>

		<section data-markdown id="taxize_example1">
			<script type="text/template">
				## Taxonomic hierarchies from NCBI/ITIS/COL/etc - <font color="red">taxize</font>

				```r
				library('taxize')
				out <- classification(c("Chironomus riparius", "Chaetopteryx"), db = "ncbi")
				lapply(out, head)
				```
<br>
				```r
				[[1]]
				                name         rank
				1 cellular organisms      no rank
				2          Eukaryota superkingdom
				3       Opisthokonta      no rank
				4            Metazoa      kingdom
				5          Eumetazoa      no rank
				6          Bilateria      no rank

				[[2]]
				                name         rank
				1 cellular organisms      no rank
				2          Eukaryota superkingdom
				3       Opisthokonta      no rank
				4            Metazoa      kingdom
				5          Eumetazoa      no rank
				6          Bilateria      no rank
				```
			</script>
		</section>

		<section data-markdown id="taxize_example2">
			<script type="text/template">
				## Get all sci. names downstream to rank X - <font color="red">taxize</font>
				```r
				col_downstream(name = "Apis", downto = "Species")
				```
				<br><br>

				```
				$Apis
				  childtaxa_id     childtaxa_name childtaxa_rank
				1      6971712 Apis andreniformis        Species
				2      6971713        Apis cerana        Species
				3      6971714       Apis dorsata        Species
				4      6971715        Apis florea        Species
				5      6971716 Apis koschevnikovi        Species
				6      6845885     Apis mellifera        Species
				7      6971717   Apis nigrocincta        Species
				```
			</script>
		</section>

		<section id="spocc1">
			<h2>Unified species occurrence data - <font color="red">spocc</font></h2>

			<img src="img/spocc_outline.png">
		</section>

		<section data-markdown id="spocc2_1">
			<script type="text/template">
			## Unified species occurrence data - <font color="red">spocc</font>

			```r
			library(spocc)
			spnames <- c('Accipiter striatus', 'Setophaga caerulescens', 'Spinus tristis')
			(out <- occ(query=spnames, from=c('gbif','ebird')))
			```
<br><br>
			```
			Summary of results - occurrences found for:
			 gbif  : 75 records across 3 species
			 bison :  0 records across 3 species
			 inat  :  0 records across 3 species
			 ebird :  75 records across 3 species
			 ecoengine :  0 records across 3 species
			 antweb :  0 records across 3 species
			```
		</script>
		</section>

		<section data-markdown id="spocc2_2">
			<script type="text/template">
			## Unified species occurrence data - <font color="red">spocc</font>

			```r
			out$gbif
			```
<br><br>
			```
			Species [Accipiter striatus (25), Setophaga caerulescens (25), Spinus tristis (25)]
			First 10 rows of [Accipiter_striatus]

			                 name  longitude latitude prov       key              issues
			1  Accipiter striatus  -72.52547 43.13234 gbif 891035349 cdround,cudc,gass84
			2  Accipiter striatus  -97.19930 32.86027 gbif 891038901 cdround,cudc,gass84
			3  Accipiter striatus  -71.72514 18.26982 gbif 891035119 cdround,cudc,gass84
			4  Accipiter striatus  -97.65347 30.15791 gbif 891040018 cdround,cudc,gass84
			5  Accipiter striatus -122.43980 37.48967 gbif 891040169 cdround,cudc,gass84
			6  Accipiter striatus  -73.06720 43.63152 gbif 891048899 cdround,cudc,gass84
			7  Accipiter striatus  -76.64497 41.85597 gbif 891043765 cdround,cudc,gass84
			8  Accipiter striatus -117.14734 32.70358 gbif 891056214 cdround,cudc,gass84
			9  Accipiter striatus  -97.88279 26.10227 gbif 891050439 cdround,cudc,gass84
			10 Accipiter striatus  -99.09873 26.49104 gbif 891049443 cdround,cudc,gass84
			..                ...        ...      ...  ...       ...                 ...
			```
			</script>
		</section>

		<section id="spocc3">
			<center><h2>Various plotting options</h2></center>
			<table cellpadding="30" cellspacing="30">
			<tr>
				<td>&nbsp;&nbsp;</td>
		<!-- <td> <div id="rotate_text">rCharts</div></td> -->
				<td>&nbsp;&nbsp;</td>
				<td> <img src="img/rcharts.png" width="550"> </td>
					<td>&nbsp;&nbsp;</td>
	 	<!-- <td> <div id="rotate_text">GitHub</div></td> -->
			 	<td>&nbsp;&nbsp;</td>
					<td> <img src="img/mapgist.png" width="550"> </td>
					<td>&nbsp;&nbsp;</td>
				</tr>
				<tr>
					<td>&nbsp;&nbsp;</td>
		<!-- <td> <div id="rotate_text">CartoDB</div></td> -->
				<td>&nbsp;&nbsp;</td>
					<td><img src="img/cartodb.png" width="550"> </td>
					<td>&nbsp;&nbsp;</td>
	 	<!-- <td> <div id="rotate_text">Shiny</div></td> -->
			 	<td>&nbsp;&nbsp;</td>
					<td><img src="img/mapshiny.png" width="550"> </td>
			</tr>
			</table>
		</section>

		<section id="viz_github">
			<h2>Visualize data interactively with GitHub</h2>
			<br>
			<script src="https://gist.github.com/sckott/7935773.js"></script>
		</section>

		<section id="viz_cartodb">
			<h2>Visualize data interactively with CartoDB</h2>
			<br>
			<iframe width='100%' height='800' frameborder='0' src='http://recology.cartodb.com/viz/76a5598e-444b-11e3-8a8c-5404a6a69006/embed_map?title=false&description=false&search=false&shareable=false&cartodb_logo=true&layer_selector=false&legends=false&scrollwheel=true&sublayer_options=1&sql=&zoom=3&center_lat=37.61423141542417&center_lon=-100.72265625'></iframe>
		</section>

		<section id="powerful">
			<br><br><br><br>
			<font size="48"><h1>this is kinda cool <i class="fa fa-smile-o"></i></h1></font>
		</section>

		<!-- ## literature section -->

		<section id="start_literature">
			<br><br><br>
			<h1><font color="blue"><i class="fa fa-table">&nbsp;Literature</font></i></h1>
		</section>

		<section data-markdown id="rplos">
			<script type="text/template">
			## Public Library of Science full text - <font color="red">rplos</font>

			```r
			library(rplos)
			plot_throughtime(list("reproducible science"), 500) +
				theme_grey(base_size = 20)
			```

			<img src="code/figure/rplos_fig.png" alt="rplos_fig" width="1100" />
			</script>
		</section>

		<section data-markdown id="rentrez_lit">
			<script type="text/template">
			## Search for articles with keyword _bioinformatics_ - <font color="red">rentrez</font>

			```r
			library('rentrez')
			res <- entrez_search(db = "pmc", term="bioinformatics")
			out <- entrez_summary(db = "pmc", id=res$ids)
			unlist(sapply(out, "[[", "DOI"))
			```
<br><br>
			```
			[1] "10.1186/1471-2164-15-772"            "10.1186/1471-2164-15-770"            "10.1186/0717-6287-47-39"
			[4] "10.1186/s12916-014-0166-6"           "10.1186/s40478-014-0093-y"           "10.1158/1541-7786.MCR-11-0421"
			[7] "10.1002/path.4294"                   "10.3978/j.issn.2218-676X.2014.06.03" "10.1186/1471-2164-15-755"
			[10] "10.1186/1471-2164-15-713"            "10.4046/trd.2014.77.2.49"            "10.4084/MJHID.2014.062"
			[13] "10.1007/s11032-014-0095-2"           "10.1186/1471-2164-15-747"            "10.1186/1471-2164-15-756"
			[16] "10.1186/1471-2164-15-733"            "10.1186/1471-2164-15-746"            "10.1186/1471-2164-15-729"
			[19] "10.1186/1471-2164-15-727"
			```
			</script>
		</section>

		<section data-markdown id="fulltext_1">
			<script type="text/template">
			## <font color="red">fulltext</font> - text mining

			```r
			library('fulltext')
			(res <- ft_search(query = 'ecology', from = c('entrez','crossref')))

			Query:
			  [ecology]
			Found:
			  [PLoS: 0; BMC: 0; Crossref: 250057; Entrez: 80815]
			Returned:
			  [PLoS: 0; BMC: 0; Crossref: 10; Entrez: 10]
			```
<br>
			```r
			print(res$entrez, n = 3)

			Query: [ecology]
			Records found, returned [80815, 10]
			    PubDate   EPubDate              Source Volume Issue Pages
			1  2012 May 2012/03/13 Am J Phys Anthropol    148     1 73-80
			2           2014/09/04     Parasit Vectors      7     1   417
			3           2014/09/02       BMC Res Notes      7     1   592
			..      ...        ...                 ...    ...   ...   ...
			Variables not shown: ArticleIds (chr), DOI (chr), FullJournalName (chr), SO (chr), Author (chr),
			     Title (chr), AuthorList (chr)
			```
			</script>
		</section>

		<section data-markdown id="fulltext_2">
			<script type="text/template">
			## <font color="red">fulltext</font> - text mining

			```r
			library('fulltext')
			library('XML')
			(res <- ft_search(query = 'ecology', from = c('entrez','plos')))
			out <- ft_get(ids = res$plos$data$id[1], from = "plos")
			text <- xmlParse(out$plos$data[[1]])
			xpathSApply(text, "//permissions")
			```
<br>
			```
			[[1]]
			<permissions>
			  <copyright-year>2013</copyright-year>
			  <copyright-holder>Carmel et al</copyright-holder>
			  <license xlink:type="simple">
			    <license-p>This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</license-p>
			  </license>
			</permissions>
			```
			</script>
		</section>


		<!-- ## altmetrics section -->

		<section id="start_altmetrics">
			<br><br><br>
			<h1><font color="black"><i class="fa fa-bar-chart-o">&nbsp;Altmetrics</font></i></h1>
		</section>

		<section id="what_altmetrics">
			<h1>article-level metrics</h1>

			<h2>There are a lot of altmetrics out there</h2>

			<img src="img/56indicatorsofimpact.png" alt="lotsalts" />
			<br>
			<a href="http://altmetrics.org/manifesto/">Canonical altmetrics document</a>
		</section>

		<section id="open_altmetrics">
			<h1>altmetrics data often open</h1>
			<br><br>
			<h2><font color="green"><i class="fa fa-unlock fa-5x"></i></font></h2>
		</section>

		<section data-markdown id="alm">
			<script type="text/template">
			## alm - analyze article metrics

			```r
			library('alm'); library('rplos'); library('plyr')
			dois <- searchplos(q='*:*', fl="id", fq=list('doc_type:full', '-article_type:correction'), limit=25)$data$id
			alm <- alm_ids(doi=dois, total_details=TRUE)
			df <- ldply(alm$data, data.frame)
			plot_density(df, source=c("counter_total","crossref_total","twitter_total","wos_total"))
			```

			<img src="img/almplot.png" alt="alm_plot" width="900" />
			</script>
		</section>

		<!-- ## pushing data up -->

		<section id="start_publish">
			<br><br><br>
			<h1><font color="green"><i class="fa fa-table">&nbsp;Publishing</font></i></h1>
		</section>

		<!-- <section id="r_publish">
			<br><br><br><br>
			<h1>R <i class="fa fa-plus"></i> data publishing <i class="fa fa-table"></i></h1>
		</section> -->

		<section id="eml1">
			<h1>EML</h1>
			<!-- <blockquote>EML (Jones et al., 2001) is a comprehensive standard that
			has been adopted by a sector of the larger international ecological research community.
			</blockquote> -->
			<br>
			<blockquote><h2>EML provides a common structure for data, to better enable ecologists to document,
			share, and interpret ecological data</h2></blockquote>
			<br><br>
			<blockquote><h2>EML standard enables data integration at the machine level (with little or no human intervention).</h2></blockquote>
<br>
			<a href="https://knb.ecoinformatics.org/#external//emlparser/docs/index.html">Read more about EML</a>
		</section>

		<section data-markdown id="eml_eg">
			<script type="text/template">
			## Valid EML

```html
<?xml version="1.0"?>
<eml:eml xmlns:eml="eml://ecoinformatics.org/eml-2.1.1"
	xmlns:ds="eml://ecoinformatics.org/dataset-2.1.1"
	xmlns:xs="http://www.w3.org/2001/XMLSchema"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:stmml="http://www.xml-cml.org/schema/stmml-1.1"
	packageId="reml_3794487.58997023" system="reml">
	<dataset>
	<title>reml example</title>
	<creator>
		<individualName>
			<givenName>Karthik</givenName>
			<surName>Ram</surName>
		</individualName>
		<electronicMailAddress>karthik.ram@gmail.com</electronicMailAddress>
	</creator>
	...
```
			</script>
		</section>

		<section id="publishing_methods">
			<h1>Publishing outlets</h1><br><br>
			<ul>
				<li><h2>Figshare</h2></li>
				<li><h2>Zenodo</h2></li>
				<li><h2>Dataone</h2></li>
				<li><h2>Various journals</h2></li>
				<li><h2>More...</h2></li>
			</ul>
		</section>

		<section id="workflow3">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<!-- Community section -->

		<section id="community">
			<font size="48"><h1>Community building</h1></font>
		</section>

		<section id="why_community">
			<h2>Why community building?</h2>
			<img src="img/2mode.jpg" alt="network" width="1100" /><br>
			<a href="http://permut.wordpress.com/2011/07/09/hbo-recycled-recycled/">link</a>
		</section>

		<section id="ropensci_community">
			<h2>The rOpenSci Community</h2>
			<img src="img/ropensci_community.png" alt="ropensci_community" width="1100"/>
			<a href="http://ropensci.org/community/#community">http://ropensci.org/community</a>
		</section>

		<section id="community_size">
			<h1>Community stats</h1><br><br>
			<ul class="b">
				<li><h2 align="left">71 contributors</h2></li>
				<li><h2 align="left">148 Github repositories</h2></li>
				<li><h2 align="left">> 9,000 commits over ~3 years</h2></li>
				<li><h2 align="left">a few pkgs with ~900 commits</h2></li>
				<li><h2 align="left">~66 R packages</h2></li>
			</ul>
		</section>

		<section id="domains">
			<h1>Domains</h1><br><br>
			<ul class="b">
				<li><h2 align="left"> <strong>Ecology</strong> </h2><br><br></li>
				<li><h2 align="left">History</h2></li>
				<li><h2 align="left">Archeology</h2></li>
				<li><h2 align="left">More...</h2></li>
			</ul>
		</section>

		<section id="training">
			<h1>Training</h1><br><br>
			<i class="fa fa-user fa-4x"></i>
			<br><br>
			<i class="fa fa-arrow-circle-down fa-4x"></i>
			<br><br>
			<i class="fa fa-graduation-cap fa-4x"></i>
<!-- <font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>
<br>
<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>
<br>
<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font>&nbsp;<font color="green"><i class="fa fa-user fa-2x"></i></font> -->
		</section>

		<!-- <section id="hackathons">
			<h1>Hackathons</h1><br><br>
			<img src="img/ropensci_hackathon.png" alt="hackathon" width="1100" />
		</section> -->

		<section id="ambassadors">
			<h1 align="left">rOpenSci Ambassador Program</h1><br><br>
			<h3><i class="fa fa-bullhorn fa-3x">Give talks</i></h3><br>
			<h3><i class="fa fa-code fa-3x">Code demos</i></h3>
		</section>

		<section id="quality">
			<h1>rOpenSci brings to the table:</h1>
			<br><br>
			<ul style="float:left;">
				<h2>Bridge between scientists and data providers</h2>
				<h2>Visibility</h2>
				<h2>Consistency</h2>
				<h2>Quality</h2>
				<h2>Conversation/community</h2>
				<h2>Stability</h2>
			</ul>
		</section>

		<section id="in_closing">
			<h1>In closing...</h1>
			<br>
			<h2>This is all about making science better</h2>
			<br>
			<h2>rOpenSci is just one vehicle with which to help</h2>
		</section>

		<section id="license">
			<br>
<h2>License: <strong>CC-BY 3.0</strong> - You are free to copy, share, adapt, or remix, photograph, film, or broadcast, blog, live-blog, or post video of this presentation, provided that you attribute the work to its author and respect the rights and licenses associated with its components.</h2>
		</section>

		<section>
			<br>
			<h3>rOpenSci on the web: <a href="http://ropensci.org/">http://ropensci.org/</a> </h3>
			<br><br>
			<h3> This talk on the web: <a href="http://recology.info/talks/uga">http://recology.info/talks/uga</a> </h3>
			<br><br><br><br><br><br>
			<h3>Made w/ <a href="https://github.com/hakimel/reveal.js">reveal.js</a></h3>
			<br>
			<h3>Icons by: <a href="http://fortawesome.github.io/Font-Awesome/">FontAwesome v4.2.0</a></h3>
		</section>
			</div>

	<!-- End slides -->


		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.min.js"></script>

		<script>
		// Full list of configuration options available here:
		// https://github.com/hakimel/reveal.js#configuration
		Reveal.initialize({
			controls: false,
			progress: true,
			history: true,
			center: false,
			slideNumber: true,
			keyboard: true,
			touch: true,

			width: 1400,
			height: 900,

			theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
			transition: Reveal.getQueryHash().transition || 'none', // default/cube/page/concave/zoom/linear/fade/none

			// Parallax scrolling
			// parallaxBackgroundImage: 'https://s3.amazonaws.com/hakim-static/reveal-js/reveal-parallax-1.jpg',
			// parallaxBackgroundSize: '2100px 900px',

			// Optional libraries used to extend on reveal.js
			dependencies: [
				{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
				{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
				{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
				{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
				{ src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
			]
		});
		</script>

	</body>
</html>
