<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>CR Talk</title>
		<meta name="description" content="SFU Talk">
		<meta name="author" content="Scott Chamberlain">
		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="css/reveal.min.css">
		<link rel="stylesheet" href="css/theme/solarized.css" id="theme">
		<link rel="stylesheet" href="css/style.css">

		<link href='http://fonts.googleapis.com/css?family=Source+Sans+Pro' rel='stylesheet' type='text/css'>

		<!-- <link rel="stylesheet" href="http://netdna.bootstrapcdn.com/bootstrap/3.0.3/css/bootstrap.min.css"> -->
		<link rel="stylesheet" href="http://netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.css">
		<link rel="stylesheet" href="lib/css/font-awesome-4.2.0/css/font-awesome.css">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">
		<!-- <link rel="stylesheet" href="lib/css/foundation.css"> -->
		<!-- <link rel="stylesheet" href="lib/css/font-awesome.min.css"> -->

		<!-- If the query includes 'print-pdf', use the PDF print sheet -->
		<script>
			document.write( '<link rel="stylesheet" href="css/print/' + ( window.location.search.match( /print-pdf/gi ) ? 'pdf' : 'paper' ) + '.css" type="text/css" media="print">' );
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->

		<style type="text/css">
			#rotate_text {
				-webkit-transform: rotate(-90deg);
				-moz-transform: rotate(-90deg);
				transform: rotate(-90deg);
				position: absolute;
				vertical-align: middle;
				filter:progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
			}

			ul.a {list-style-type: circle; margin: 0; padding: 0;}
			ul.b {list-style-type: square; margin-left: 0; padding-left: 0;}
		</style>
	</head>

	<body>

		<div class="reveal">

	<!-- Any section element inside of this container is displayed as a slide -->
	<div class="slides">
		<section id="intro">
				<img src="img/ropensci_main.png" alt="ropensci" />
				<h2><font color="#B8172F">Tools for Reproducible Research</font></h2><br>
				<p class="grey">Scott Chamberlain (@ropensci)
					<p class="grey">UC Berkeley / rOpenSci
				<!-- <p class="grey">recology.info/talks/sfustats<br><br> --><br><br>
				<center>Supported by:</center>
				<img src="img/sloan_logo2.png" alt="sloan" />
		</section>

		<section id="link">
			<h2> This talk on the web: <br><br><a href="http://recology.info/talks/cr-talk">recology.info/talks/cr-talk</a> </h2>
		</section>

		<section id="license">
			<br>
<h2>License: <strong>CC-BY 3.0</strong> - You are free to copy, share, adapt, or remix, photograph, film, or broadcast, blog, live-blog, or post video of this presentation, provided that you attribute the work to its author and respect the rights and licenses associated with its components.</h2>
		</section>

		<section id="notes">
			<center><img src="img/notesfromnature.png" width="1100"></center>
		</section>

		<section id="sensors">
			<center><img src="img/sensors.png" width="1100"></center>
		</section>

		<section id="manual_data">
			<center><img src="img/ropensci_data.png" width="1100"></center>
		</section>

		<section id="messy_data">
			<center><img src="img/esa_archives.png" width="1200"></center>
		</section>

		<section id="data_hard">
			<h1 align="left">Huge amount of open data!</h1>
			<h2><i class="fa fa-database fa-5x" style="color:#008000"></i></h2>
		</section>

		<section id="we_need">
			<center><h1 align="left">But...we aren't trained to use it</h2></center>
				<br><br>
				<center><div class="row">
					<h2><i class="fa fa-ban fa-5x" style="color:#b22d28"></i></font></h2>
				</div></center>
		</section>

		<section id="why_reproducible_selfish">
			<font color="blue"><i class="fa fa-refresh fa-3x">&nbsp;Why Reproducible?</i> </font>
			<br><br><br><br>
			<h2>For yourself!</h2>
			<br><br>
			<i class="fa fa-user fa-5x"></i>
			<!-- <h2 style="float:left;"> <i class="fa fa-code"></i> for others (if reproducible, more confidence in results) </h2> -->
		</section>

		<section id="why_reproducible_mistakes">
			<font color="blue"><i class="fa fa-refresh fa-3x">&nbsp;Why Reproducible?</i> </font>
			<br><br><br>
			<h2>To avoid mistakes</h2>
			<br><br>
			<table cellpadding="40" cellspacing="40">
			<tr>
			<td>&nbsp;&nbsp;</td>
				<td>		<center><p>Reinhart-Rogoff<img src="img/excel.png" width="500"><br>
				<a href="http://www.nytimes.com/2013/04/19/opinion/krugman-the-excel-depression.html?_r=0">NYTimes</a></center></td>
			<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td>
				<td>		<center><p>Amgen Reproduciblity Study<br><br><i class="fa fa-flask fa-5x"></i></center><br>
				<a href="http://www.reuters.com/article/2012/03/28/us-science-cancer-idUSBRE82R12P20120328" style="float:left;">Reuters</a>&nbsp;--&nbsp;
				<a href="http://www.economist.com/news/briefing/21588057-scientists-think-science-self-correcting-alarming-degree-it-not-trouble">The Economist</a></td>
			</tr>
			</table>
		</section>

		<!-- <section id="easier">
			<font size="52"><h1 align="left">Or maybe you just want science to be easier</h1></font>
		</section> -->

		<section id="whats_needed">
			<font size="52"><h1 align="left">What's needed?</h1></font>
		</section>

		<section id="toolset">
			<br>
			<img src="img/tools.png" alt="toolset" width="1200" />
		</section>

		<section id="reproducible_workflow">
			<font size="52"><h1 align="left"><i class="fa fa-arrow-circle-right"></i>&nbsp;A reproducible workflow</h1></font>
		</section>

		<section id="these">
			<center><h2>These!</h2></center>
			<img src="img/opentools.png" width="1000">
		</section>

		<section id="learning_curve">
			<center><h2>There's a learning curve, but...</h2></center>
			<img src="img/tasks.jpg" width="1000">
			<a href="http://www.globalnerdy.com/2012/04/24/geeks-and-repetitive-tasks/">link</a>
		</section>

		<section id="workflows">
		<center><h2> Workflows side by side</h2></center>
			<table cellpadding="40" cellspacing="20">
				<tr>
	<td>&nbsp;&nbsp;</td>
		<td width="50%"> <h3>Our workflow now</h3>
				<ul>
				<font size="8">
					<li>Browser</li>
					<li>Excel</li>
					<li>SAS</li>
					<li>SigmaPlot</li>
					<li>Word</li>
					<li>Endnote</li>
				</font>
				</ul> </td>
		<td>&nbsp;&nbsp;</td>

		 <td width="50%">	<h3>What it could be</h3>
			<ul>
				<font size="8">
				<li>R (or Python)</li>
				<li>+ a few open source things (<a href="http://daringfireball.net/projects/markdown/">Markdown</a>, or <a href="http://www.latex-project.org/">LaTeX</a> )</li></ul></td>
			</font>
	<td>&nbsp;&nbsp;</td>
		</tr>
		<tr>
<td>&nbsp;&nbsp;</td>
				<td><font size="8"> <i class="fa fa-money"></i> Cost = $$$$$$$<br>
					<p> <i class="fa fa-unlock"></i> Open? = Nope
					<p> <i class="fa fa-refresh"></i> Reproducible? = Nope</font></td>
<td>&nbsp;&nbsp;</td>
				<td><font size="8"> <i class="fa fa-money"></i> Cost = 0<br>
					<p> <i class="fa fa-unlock"></i> Open? = Yes!
					<p> <i class="fa fa-refresh"></i> Reproducible? = Yep</font></td>

		</tr>
	</table>
		</section>

		<section id="perfect">
			<center><h2>A perfect combination</h2></center>
			<br><br>
			<center>
				<font color="red" size="12"><i class="fa fa-refresh fa-5x">&nbsp;</i></font>
				<font color="green" size="12"> <i class="fa fa-unlock fa-5x">&nbsp;</i></font>
				<font color="blue" size="12" class="fragment"><i class="fa fa-cloud fa-5x">&nbsp;</i></font>
			</center>
			<br>
			<h2 class="fragment">Data is increasingly on the web <font color="blue"><i class="fa fa-cloud"></i></font></h2>
		</section>

		<!-- <section id="dataweb">
			<h2>Data is increasingly on the web <font color="blue"><i class="fa fa-cloud"></i></font></h2>
			<img src="img/data.png" width="1000">
		</section> -->

		<section id="apis1">
			<br><br>
			<h2>API: Application Programming Interface</h2>
			<br><br><br><br>
			<h2><i class="fa fa-link fa-3x"></i>&nbsp;&nbsp;</h2>
		</section>

		<section id="apis2">
			<br><br>
			<h2> <font color="red">Reproducibly</font> plug data from the web into research workflows</h2>
		</section>

		<section id="ropensci">
			<br><br><br>
			<center><img src="img/ropensci_main.png"></center>
			<br><br>
			<center><img src="img/sloan_logo2.png"></center>
		</section>

		<section id="ropensci_team">
			<h2>How rOpenSci got started</h2>
			    <div>
			    <img src="img/core_team.png" width="100%">
			    </div>
					<br><br>
			    <h3>formed from ad-hoc conversation over Twitter. Now a worldwide community of researchers <br><br> <a href="http://ropensci.org/community/#community" title="rOpenSci Community" target="_blank">http://ropensci.org/community</a></h3>
		</section>

		<section id="cranpkgs">
			<h2>R has a lot of packages<br>(mostly data manip./viz & statistics)</h2>
			<br>
			<!-- library("ggplot2")
			df <- data.frame(date = as.Date(c('2014-10-29','2013-11-08','2012-08-23','2011-05-12',
			                                  '2009-10-04','2007-04-12','2004-10-01','2003-04-01')),
			                 pkgs = c(6000, 5000, 4000, 3000, 2000, 1000, 500, 250))
			ggplot(df, aes(date, pkgs)) + geom_line(size=3) + theme_grey(base_size = 26) -->
			<img src="img/rpkgs.png" width="90%">
		</section>

		<!-- <section id="ropensci_does">

		</section> -->

		<section id="workflow">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_acquisition">
			<br><br><br>
			<center>
				<h2><font color="blue"><strong>Data acquisition</strong> <i class="fa fa-cloud-download"></i></font> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i>&nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_manip">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> <font color="blue"><strong>data manipulation/analysis/viz</strong> <i class="fa fa-bar-chart"></i></font> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_writing">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> <font color="blue"><strong>writing</strong> <i class="fa fa-pencil"></i></font> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> publish <i class="fa fa-cloud-upload"></i></h2>
			</center>
		</section>

		<section id="workflow_publish">
			<br><br><br>
			<center>
				<h2>Data acquisition <i class="fa fa-cloud-download"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> data manipulation/analysis/viz <i class="fa fa-bar-chart"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> writing <i class="fa fa-pencil"></i> &nbsp;&nbsp;&nbsp; <i class="fa fa-plus"></i> <br><br> <font color="blue"><strong>publish</strong> <i class="fa fa-cloud-upload"></i></font></h2>
			</center>
		</section>

		<section id="packages">
			<h2>rOpenSci packages</h2>

			<h3><a href="http://ropensci.org/packages/">ropensci.org/packages</a> </h3>

			<table cellpadding="20" cellspacing="20">
				<tr>
	<!-- <td>&nbsp;</td> -->
		<td>	<h3> <font color="red"> <i class="fa fa-flask">&nbsp;Biology</i> </font> </h3>
			<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/taxize">taxize</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rgbif">rgbif</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rbison">rbison</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/treebase">treebase</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rfishbase">rfishbase</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/RNeXML">RNeXML</a></h3> </li>
				</ul>
		</td>
		<td>&nbsp;</td>
		<td>&nbsp;</td>
		<td>&nbsp;</td>
		<td>&nbsp;</td>
		<td>
				<h3> <font color="blue"><i class="fa fa-book">&nbsp;Literature</font></i> </h3>
				<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/rplos">rplos</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rmetadata">rmetadata</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rentrez">rentrez</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/fulltext">fulltext</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/bmc">bmc</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/elife">eLife</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rcrossref">rcrossref</a></h3> </li>
				</ul>
		</td>
		<td>&nbsp;</td>
		<td>&nbsp;</td>
		<td>&nbsp;</td>
			<td>
					<h3><font color="green"><i class="fa fa-bars">&nbsp;Publishing</font></i></h3>
					<br>
					<ul>
						<li> <h3><a href="https://github.com/ropensci/rfigshare">rfighsare</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/EML">EML</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/zenodo">zenodo</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/dvn">dvn</a></h3> </li>
						<li> <h3><a href="https://github.com/ropensci/aRxiv">aRxiv</a></h3> </li>
					</ul>
			</td>
			</table>
		</section>

		<section id="other_fields">
			<h2>rOpenSci packages</h2>

			<h3><a href="http://ropensci.org/packages/">ropensci.org/packages</a> </h3>

			<table cellpadding="20" cellspacing="20">
				<tr>
	<!-- <td>&nbsp;</td> -->
		<td>	<h3> <font color="orange"> <i class="fa fa-history">&nbsp;History</i> </font> </h3>
			<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/rdpla">rdpla</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/reuropeana">reuropeana</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/gender">gender</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/historydata">historydata</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/usboundaries">usboundaries</a></h3> </li>
				</ul>
		</td>
		<td>&nbsp;</td>
		<td>
				<h3> <font color="purple"><i class="fa fa-university">&nbsp;Archeology</font></i> </h3>
				<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/pleiades">pleiades</a></h3> </li>
				</ul>
	</td>
	<td>&nbsp;</td>
		<td>
		<h3> <font color="black"><i class="fa fa-bar-chart-o">&nbsp;Altmetrics</font></i> </h3>
				<br>
				<ul>
					<li> <h3><a href="https://github.com/ropensci/alm">alm</a></h3> </li>
					<li> <h3><a href="https://github.com/ropensci/rAltmetric">rAltmetric</a></h3> </li>
				</ul>
		</td>
			</table>
		</section>

		<section id="data_growth">
			<h2>Some numbers for amount of the data</h2>
			<table class="table table-bordered table-striped">
			 <thead>
			  <tr>
			   <th style="text-align:left;"> source </th>
			   <th style="text-align:right;"> value </th>
			   <th style="text-align:left;"> type </th>
			  </tr>
			 </thead>
			<tbody>
			  <tr>
			   <td style="text-align:left;"> GBIF </td>
			   <td style="text-align:right;"> 420222471 </td>
			   <td style="text-align:left;"> Species occurrence records </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> BHL </td>
			   <td style="text-align:right;"> 155891133 </td>
			   <td style="text-align:left;"> Names </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> BHL </td>
			   <td style="text-align:right;"> 43968949 </td>
			   <td style="text-align:left;"> Pages </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> DataCite </td>
			   <td style="text-align:right;"> 3618096 </td>
			   <td style="text-align:left;"> Data records </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> eBird </td>
			   <td style="text-align:right;"> 2923886 </td>
			   <td style="text-align:left;"> Observations </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> NPN </td>
			   <td style="text-align:right;"> 2537095 </td>
			   <td style="text-align:left;"> Phenology records </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> Neotoma </td>
			   <td style="text-align:right;"> 2200221 </td>
			   <td style="text-align:left;"> Data records </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> OpenSNP </td>
			   <td style="text-align:right;"> 2140939 </td>
			   <td style="text-align:left;"> SNPs </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> COL </td>
			   <td style="text-align:right;"> 1352112 </td>
			   <td style="text-align:left;"> Taxonomic names </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> ITIS </td>
			   <td style="text-align:right;"> 624282 </td>
			   <td style="text-align:left;"> Taxonomic names </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> eBird </td>
			   <td style="text-align:right;"> 205970 </td>
			   <td style="text-align:left;"> Checklist records </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> BHL </td>
			   <td style="text-align:right;"> 139561 </td>
			   <td style="text-align:left;"> Items </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> PLOS </td>
			   <td style="text-align:right;"> 136330 </td>
			   <td style="text-align:left;"> Articles </td>
			  </tr>
			  <tr>
			   <td style="text-align:left;"> BHL </td>
			   <td style="text-align:right;"> 77258 </td>
			   <td style="text-align:left;"> Titles </td>
			  </tr>
			</tbody>
			</table>
		</section>

		<section id="demos">
			<h1>Code demos</h1>
			<br><br>
			<ul>
				<h2>The diversity of projects</h2>
				<h2>How easy it is</h2>
				<h2>Imagine these examples at scale</h2>
			</ul>
		</section>

		<!-- ## literature section -->
		<section id="start_literature">
			<br><br><br>
			<h1><font color="blue"><i class="fa fa-book">&nbsp;Literature</font></i></h1>
		</section>

		<section data-markdown id="rplos">
			<script type="text/template">
			## Public Library of Science full text - <font color="red">rplos</font>

			```r
			library(rplos)
			plot_throughtime(list("reproducible science"), 500) +
				theme_grey(base_size = 20)
			```

			<img src="code/figure/rplos_fig.png" alt="rplos_fig" width="1100" />
			</script>
		</section>

		<section data-markdown id="rentrez_lit">
			<script type="text/template">
			## Search for articles with keyword _bioinformatics_ - <font color="red">rentrez</font>

			```r
			library('rentrez')
			res <- entrez_search(db = "pmc", term="bioinformatics")
			out <- entrez_summary(db = "pmc", id=res$ids)
			unlist(sapply(out, "[[", "DOI"))
			```
<br><br>
			```
			[1] "10.1186/1471-2164-15-772"            "10.1186/1471-2164-15-770"            "10.1186/0717-6287-47-39"
			[4] "10.1186/s12916-014-0166-6"           "10.1186/s40478-014-0093-y"           "10.1158/1541-7786.MCR-11-0421"
			[7] "10.1002/path.4294"                   "10.3978/j.issn.2218-676X.2014.06.03" "10.1186/1471-2164-15-755"
			[10] "10.1186/1471-2164-15-713"            "10.4046/trd.2014.77.2.49"            "10.4084/MJHID.2014.062"
			[13] "10.1007/s11032-014-0095-2"           "10.1186/1471-2164-15-747"            "10.1186/1471-2164-15-756"
			[16] "10.1186/1471-2164-15-733"            "10.1186/1471-2164-15-746"            "10.1186/1471-2164-15-729"
			[19] "10.1186/1471-2164-15-727"
			```
			</script>
		</section>

		<section data-markdown id="fulltext_1">
			<script type="text/template">
			## <font color="red">fulltext</font> - text mining

			```r
			library('fulltext')
			(res <- ft_search(query = 'ecology', from = c('entrez','crossref')))

			Query:
			  [ecology]
			Found:
			  [PLoS: 0; BMC: 0; Crossref: 250057; Entrez: 80815]
			Returned:
			  [PLoS: 0; BMC: 0; Crossref: 10; Entrez: 10]
			```
<br>
			```r
			print(res$entrez, n = 3)

			Query: [ecology]
			Records found, returned [80815, 10]
			    PubDate   EPubDate              Source Volume Issue Pages
			1  2012 May 2012/03/13 Am J Phys Anthropol    148     1 73-80
			2           2014/09/04     Parasit Vectors      7     1   417
			3           2014/09/02       BMC Res Notes      7     1   592
			..      ...        ...                 ...    ...   ...   ...
			Variables not shown: ArticleIds (chr), DOI (chr), FullJournalName (chr), SO (chr), Author (chr),
			     Title (chr), AuthorList (chr)
			```
			</script>
		</section>

		<section data-markdown id="fulltext_2">
			<script type="text/template">
			## <font color="red">fulltext</font> - text mining

			```r
			library('fulltext')
			library('XML')
			(res <- ft_search(query = 'ecology', from = c('entrez','plos')))
			out <- ft_get(ids = res$plos$data$id[1], from = "plos")
			text <- xmlParse(out$plos$data[[1]])
			xpathSApply(text, "//permissions")
			```
<br>
			```
			[[1]]
			<permissions>
			  <copyright-year>2013</copyright-year>
			  <copyright-holder>Carmel et al</copyright-holder>
			  <license xlink:type="simple">
			    <license-p>This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</license-p>
			  </license>
			</permissions>
			```
			</script>
		</section>

		<!-- ## pushing data up -->

		<section id="start_publish">
			<br><br><br>
			<h1><font color="green"><i class="fa fa-bars">&nbsp;Publishing</font></i></h1>
		</section>

		<!-- <section id="r_publish">
			<br><br><br><br>
			<h1>R <i class="fa fa-plus"></i> data publishing <i class="fa fa-table"></i></h1>
		</section> -->

		<section id="eml1">
			<img src="img/carl.jpg" alt="carl" width="10%" style="float: right;" />
			<h1>EML</h1>
			<!-- <blockquote>EML (Jones et al., 2001) is a comprehensive standard that
			has been adopted by a sector of the larger international ecological research community.
			</blockquote> -->
			<br>
			<blockquote><h2>EML provides a common structure for data, to better enable ecologists to document,
			share, and interpret ecological data</h2></blockquote>
			<br><br>
			<blockquote><h2>EML standard enables data integration at the machine level (with little or no human intervention).</h2></blockquote>
<br>
			<a href="https://knb.ecoinformatics.org/#external//emlparser/docs/index.html">Read more about EML</a>
		</section>

		<section data-markdown id="eml_eg">
			<script type="text/template">
			## Valid EML

```html
<?xml version="1.0"?>
<eml:eml xmlns:eml="eml://ecoinformatics.org/eml-2.1.1"
	xmlns:ds="eml://ecoinformatics.org/dataset-2.1.1"
	xmlns:xs="http://www.w3.org/2001/XMLSchema"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:stmml="http://www.xml-cml.org/schema/stmml-1.1"
	packageId="reml_3794487.58997023" system="reml">
	<dataset>
	<title>reml example</title>
	<creator>
		<individualName>
			<givenName>Karthik</givenName>
			<surName>Ram</surName>
		</individualName>
		<electronicMailAddress>karthik.ram@gmail.com</electronicMailAddress>
	</creator>
	...
```
			</script>
		</section>

		<section id="publishing_methods">
			<h1>Publishing outlets</h1><br><br>
			<ul>
				<li><h2>Figshare</h2></li>
				<li><h2>Zenodo</h2></li>
				<li><h2>Dataone</h2></li>
				<li><h2>Various journals</h2></li>
				<li><h2>More...</h2></li>
			</ul>
		</section>
		<!-- end publishing section -->

		<!-- altmetrics section -->
		<section id="start_altmetrics">
			<br><br><br>
			<h1><font color="black"><i class="fa fa-bar-chart-o">&nbsp;Altmetrics</font></i></h1>
		</section>

		<section id="what_altmetrics">
			<h1>article-level metrics</h1>

			<h2>There are a lot of altmetrics out there</h2>

			<img src="img/56indicatorsofimpact.png" alt="lotsalts" />
			<br>
			<a href="http://altmetrics.org/manifesto/">Canonical altmetrics document</a>
		</section>

		<section id="open_altmetrics">
			<h1>altmetrics data often open</h1>
			<br><br>
			<h2><font color="green"><i class="fa fa-unlock fa-5x"></i></font></h2>
		</section>

		<section data-markdown id="alm">
			<script type="text/template">
			## alm - analyze article metrics

			```r
			library('alm'); library('rplos'); library('plyr')
			dois <- searchplos(q='*:*', fl="id", fq=list('doc_type:full', '-article_type:correction'), limit=25)$data$id
			alm <- alm_ids(doi=dois, total_details=TRUE)
			df <- ldply(alm$data, data.frame)
			plot_density(df, source=c("counter_total","crossref_total","twitter_total","wos_total"))
			```

			<img src="img/almplot.png" alt="alm_plot" width="900" />
			</script>
		</section>
		<!-- end altmetrics section -->

		<section id="end">
			<br>
			<h3>rOpenSci on the web: <a href="http://ropensci.org/">ropensci.org</a> </h3>
			<br><br>
			<h3> This talk on the web: <a href="http://recology.info/talks/cr-talk">recology.info/talks/cr-talk</a> </h3>
			<br><br><br><br><br><br>
			<h3>Made w/ <a href="https://github.com/hakimel/reveal.js">reveal.js</a></h3>
			<br>
			<h3>Icons by: <a href="http://fortawesome.github.io/Font-Awesome/">FontAwesome v4.2.0</a></h3>
		</section>

			</div>

	<!-- End slides -->


		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.min.js"></script>

		<script>
		// Full list of configuration options available here:
		// https://github.com/hakimel/reveal.js#configuration
		Reveal.initialize({
			controls: false,
			progress: true,
			history: true,
			center: false,
			slideNumber: true,
			keyboard: true,
			touch: true,

			width: 1400,
			height: 900,

			theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
			transition: Reveal.getQueryHash().transition || 'none', // default/cube/page/concave/zoom/linear/fade/none

			// Parallax scrolling
			// parallaxBackgroundImage: 'https://s3.amazonaws.com/hakim-static/reveal-js/reveal-parallax-1.jpg',
			// parallaxBackgroundSize: '2100px 900px',

			// Optional libraries used to extend on reveal.js
			dependencies: [
				{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
				{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
				{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
				{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
				{ src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
			]
		});
		</script>

	</body>
</html>
